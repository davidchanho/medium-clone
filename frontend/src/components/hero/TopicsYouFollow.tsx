import React from "react";
import { Button, ListGroup } from "react-bootstrap";
import TopicList from "../topic-list";
import { IconPlus } from "../_shared/icons";

function TopicsYouFollow() {
  return (
    <div>
      <p className="text-uppercase font-weight-bold mb-2">Topics you follow</p>
      <ListGroup variant="flush">
        <TopicList />
        <Button variant="light" size="sm" className="p-2 h-50">
          <IconPlus />
        </Button>
      </ListGroup>
    </div>
  );
}

export default TopicsYouFollow;
