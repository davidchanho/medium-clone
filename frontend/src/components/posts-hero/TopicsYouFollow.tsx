import React from "react";
import TopicList from "../topic-list";

function TopicsYouFollow() {
  return (
    <div>
      <p>Topics you follow</p>
      <ul>
        <TopicList />

        <a href="#!">See More</a>
      </ul>
    </div>
  );
}

export default TopicsYouFollow;
