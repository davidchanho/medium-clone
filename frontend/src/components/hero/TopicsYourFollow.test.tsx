import React from "react";
import renderer from "react-test-renderer";
import TopicsYouFollow from "./TopicsYouFollow";

describe("TopicsYouFollow", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TopicsYouFollow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
