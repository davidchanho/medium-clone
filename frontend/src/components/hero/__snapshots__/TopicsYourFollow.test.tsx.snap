// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicsYouFollow renders correctly 1`] = `
<div>
  <p
    className="text-uppercase font-weight-bold mb-2"
  >
    Topics you follow
  </p>
  <div
    className="list-group list-group-flush"
    onKeyDown={[Function]}
  >
    <div
      className="mr-2 mb-2 list-group-item list-group-item-light"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      topic1
    </div>
    <div
      className="mr-2 mb-2 list-group-item list-group-item-light"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      topic2
    </div>
    <div
      className="mr-2 mb-2 list-group-item list-group-item-light"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      topic3
    </div>
    <button
      className="p-2 h-50 btn btn-light btn-sm"
      disabled={false}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M8 3.5a.5.5 0 01.5.5v4a.5.5 0 01-.5.5H4a.5.5 0 010-1h3.5V4a.5.5 0 01.5-.5z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M7.5 8a.5.5 0 01.5-.5h4a.5.5 0 010 1H8.5V12a.5.5 0 01-1 0V8z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
